sorts/tree_sort.ts

```diff
@@ -7,54 +7,12 @@
  * @see <https://en.wikipedia.org/wiki/Tree_sort>
  */
 
-type BinaryTree<T> = {
-  leftSubTree: BinaryTree<T> | null;
-  node: T | null;
-  rightSubTree: BinaryTree<T> | null;
-};
+import { BinarySearchTree } from "../data_structures/tree/binary_search_tree";
 
 export const treeSort = (arr: number[]): number[] => {
```

Comment:
Please make this generic over the type of `arr`; this can be used to sort anything comparable (such as strings, for example).